source/routing/raptor_api.cpp

```diff
@@ -1369,97 +1370,87 @@ void filter_late_journeys(RAPTOR::Journeys& journeys, const NightBusFilter::Para
-bool shorten_section(navitia::routing::Journey::Section& section,
-                     const std::string& stop_area_uri,
-                     const navitia::type::VehicleJourney* vj_to_skip,
-                     const map_stop_point_duration& departures,
-                     const map_stop_point_duration& destinations,
-                     const bool clockwise) {
+bool shorten_section_clockwise(navitia::routing::Journey::Section& section,
+                               const std::string& last_stop_area_uri,
+                               const navitia::type::VehicleJourney* last_vj,
+                               const map_stop_point_duration& fallbacks) {
     log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+
+    // because of stay-ins, we may have several vj in one section, we have to scan the stop times of all vjs
     auto order = section.get_in_st->order();
-    // because of stay-ins, we may have several vj in one section, we have to scan the stop times
-    // of all vjs
     for (const auto* vj = section.get_in_st->vehicle_journey; vj; (vj = vj->next_vj, order = type::RankStopTime(0))) {
+        if (vj == last_vj) {
+            return false;
+        }
+
+        if (section.get_in_dt < section.get_in_st->boarding_time) {
+            LOG4CPLUS_ERROR(logger, "Error : section.get_in_dt < section.get_in_st->boarding_time");
+            return false;
+        }
+        // determine midnigth of the day at which the vj is used
+        // WARNING : this base_dt may need to be increased in case there is some stay-ins vehicle
+        // with stop_times that are past midnight (not handled yet)
+        navitia::DateTime base_dt = section.get_in_dt - section.get_in_st->boarding_time;
         for (const auto& st :
              boost::make_iterator_range(vj->stop_time_list.begin() + order.val, vj->stop_time_list.end())) {
-            if (can_shorten_at(departures, destinations, st, clockwise)
-                && st.stop_point->stop_area->uri == stop_area_uri && vj != vj_to_skip) {
-                (clockwise ? section.get_out_st : section.get_in_st) = &st;
-                (clockwise ? section.get_out_dt : section.get_in_dt) = clockwise ? st.arrival_time : st.departure_time;
+            // check stop_area is the same and it's possible to use stop_point to get out of PT
+            if (st.stop_point->stop_area->uri == last_stop_area_uri
+                && navitia::contains(fallbacks, routing::SpIdx{st.stop_point->idx}) && st.drop_off_allowed()) {
+                section.get_out_st = &st;
+                section.get_out_dt = base_dt + st.alighting_time;
 
                 return true;
             }
-            if (section.get_out_st == &st) {
-                return false;
-            }
         }
     }
-    LOG4CPLUS_WARN(logger, "Error occurred when modifying backtracking journeys");
+    LOG4CPLUS_ERROR(logger, "Error occurred when modifying backtracking journeys clockwise");
```

Comment:
This is not an error anymore to arrive here. You will reach the end of the function on every section (apart the last one) that cannot be shortened

Reply:
As discussed: Well-seen, only the last section of the VJ uses last_vj, so this case would happen regularly.
But then on intermediate sections we could search too far and create bugs.

So chose to keep it an error message and instead add a test that we reached section's end stop-time: https://github.com/hove-io/navitia/pull/3817/commits/4c3e911487e0d5193e2421e0c3c4546b6638e01c

Reply:
:sweat_smile: Actually, stop shortening serach AFTER checking the end stop-time: https://github.com/hove-io/navitia/pull/3817/commits/3bb9c1374a4309b540560449de066d2b9b7d0a1d